### PR TITLE
[commonmark-react-renderer] Add types

### PR DESCRIPTION
**Pull request by Methuselah96**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/commonmark-react-renderer/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            renderer: string;
+            props: unknown;
+            children: React.ReactNode[];
+        }) => unknown)
+            | undefined;
+        allowedTypes: string[];
+        unwrapDisallowed: boolean;
+        render: (root: Node) => React.ReactNode;
+        linkTarget: string | false;
+    };
```
Comment (sheetalkamat) [inline types/commonmark-react-renderer/index.d.ts:120]:
```suggestion
    new (options?: ReactRenderer.Options): Required<ReactRenderer.Options>;
```

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/commonmark-react-renderer/index.d.ts (hunk 1 above)
Comment (Methuselah96) [reply types/commonmark-react-renderer/index.d.ts:120]: Hey, thanks for the review! The values returned are not the same as options so this wouldn't work.

Comment (sheetalkamat) [reply types/commonmark-react-renderer/index.d.ts:120]: If thats the case i recommend creating name for this interface just like options so its easier to read,

Comment (Methuselah96) [reply types/commonmark-react-renderer/index.d.ts]: @sheetalkamat Should be good now.

**Pull request merged**